pandas/core/generic.py

```diff
@@ -2012,6 +2012,10 @@ def sample(self, n=None, frac=None, replace=False, weights=None, random_state=No
         # Check weights for compliance
         if weights is not None:
 
+            # If a series, align with frame
+            if isinstance(weights, pd.Series):
```

Comment:
Added, though I stuck with the alignment default of not filling gaps with NaNs -- since the documentation already says series must be same length as object being sampled, this felt more consistent.